src/test/analyzer_test.ts

```diff
@@ -532,6 +532,45 @@ suite('Analyzer', () => {
     });
   });
 
+  test('analyzes a document with a namespace', async() => {
+    const document =
+        await analyzer.analyze('static/namespaces/import-all.html');
+
+    const namespaces =
+        Array.from(document.getByKind('namespace', {imported: true}));
+    assert.deepEqual(namespaces.map((b) => b.name), [
+      'ExplicitlyNamedNamespace',
+      'ExplicitlyNamedNamespace.NestedNamespace',
+      'ImplicitlyNamedNamespace',
+      'ImplicitlyNamedNamespace.NestedNamespace',
+      'DynamicNamespace.ArrayNotation',
+      'DynamicNamespace.Aliased',
+    ]);
+  });
+
+  test('creates warnings when duplicate namespaces are analyzed', async() => {
+    const document =
+        await analyzer.analyze('static/namespaces/import-duplicates.html');
+    const namespaces =
+        Array.from(document.getByKind('namespace', {imported: true}));
+    assert.deepEqual(namespaces.map((b) => b.name), [
+      'ExplicitlyNamedNamespace',
+      'ExplicitlyNamedNamespace.NestedNamespace',
+    ]);
+    assert.deepEqual(
+        document.getWarnings({imported: true}), [{
+          message:
+              'Found more than one namespace named ExplicitlyNamedNamespace.',
+          severity: Severity.WARNING,
+          code: 'multiple-polymer-namespaces',
+          sourceRange: {
+            file: 'static/namespaces/namespace-duplicate.js',
+            start: {column: 0, line: 5},
```

Comment:
Here and elsewhere in this PR, asserting on source ranges is really brittle. If we use CodeUnderliner the code is easier to read, easier to maintain, and isn't broken by adding unrelated lines to a fixture file.

You can use assertSubset to leave out the sourceRange property of the expected object.